vertical-pod-autoscaler/recommender/model/types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package model
+
+import (
+	"time"
+)
+
+// MeasuredResourceName represent the name of the resource monitored by recommender
+type MeasuredResourceName string
+
+const (
+	// ResourceCPU represents CPU in milicores (1core = 1000milicores)
+	ResourceCPU MeasuredResourceName = "cpu"
+	// ResourceMemory represents memory, in bytes. (500Gi = 500GiB = 500 * 1024 * 1024 * 1024)
+	ResourceMemory MeasuredResourceName = "memory"
+)
+
+// PodID contains information needed to identify a Pod within a cluster
+type PodID struct {
+	// Namespaces where the Pod is defined
+	Namespace string
+	// PodName is a name of the pod unique within a namespace
+	PodName string
+}
+
+// ContainerID contains information needed to identify a Container within a cluster
+type ContainerID struct {
+	PodID
+	// ContainerName is the name of the container, unique within a pod.
+	ContainerName string
+}
+
+// ContainerMetricsSnapshot contains information about usage of certain container withing defined time window
+type ContainerMetricsSnapshot struct { // nazwa zostaje
+	// ID identifies a specific container those metrics are comming from
+	ID ContainerID
+	// Time when the snapshot was taken
+	SnapshotTime time.Time
+	// duration, which this snapshot represents
+	SnapshotWindow time.Duration
+	// Actual usage of resources, during the SnapshotWindow.
```

Comment:
"Average usage of the resources over the measurement interval"

Reply:
Good point, however, I was not able to find this information. Metrics api contains just types definitions and does not provide aggregation method, and API server, which we use, is pretty un-documented. 
(As a reference, heapster was using avg for cpu and current consumption for memory, but it is different mechanism here)